File: GeneticAlgorithm/output/att48_fit.py
import matplotlib.pyplot as plt
import numpy as np

data = np.loadtxt('att48_roulette.f')
data = np.transpose(data)

plt.figure(1)
plt.plot(data[0, :], data[1, :], 'r')
plt.plot(data[0, :], data[2, :], 'b')
plt.legend(['Fitness of best solution', 'Average population fitness'])
plt.savefig('plots/att48_fit.png')
plt.figure(2)
plt.loglog(data[0, :], data[1, :], 'r')
plt.loglog(data[0, :], data[2, :], 'b')
plt.legend(['Fitness of best solution', 'Average population fitness'])
plt.savefig('plots/att48_fit_log.png')

File: GeneticAlgorithm/output/att48_fit.m
a = importdata('att48_roulette.f');
b = a.data;

generation = b(:, 1);
fitness = b(:, 2);
avgfitness = b(:, 3);

figure(1)
plot(generation, fitness, 'r', generation, avgfitness, 'b')
xlim([0 max(generation)])
legend('Fitness of best solution','Average population fitness')
saveas(1, 'plots/att48_fit_roulette.png')
figure(2)
loglog(generation, fitness, 'r', generation, avgfitness, 'b')
xlim([0 max(generation)])
legend('Fitness of best solution','Average population fitness')
saveas(1, 'plots/att48_fit_log_roulette.png')
